### PR TITLE
Maroon-504 Updated NCalc package

### LINKED ISSUE #504: Update NCalc package
Updating NCalc package to 5.2.0 in order to fix some bugs. E.g. calculation of abs(x)/0.1 throws an error in older version.

### PR DESCRIPTION
**Pull request by flowmdev**
Closes #504

### CODE REVIEW

File: unity/Assets/Vendor/ncalc/LICENSE
Comment (FlorianGlawogger) [inline unity/Assets/Vendor/ncalc/LICENSE:1]: why has the license file been removed?

**Review by FlorianGlawogger — CHANGES_REQUESTED** [review]
Seems to work fine, I think we should still include License files though? 
(I merged the PR with the develop branch)

**Comment by jonathanplatzer** [comment]
When using this there is a regression in the Motion Solver, it seems like the newer versions of NCalc changed how functions and parameters are evaluated, so defining own functions like ``h(x): if(x == 0, 0.5, (1 + x/abs(x))/2)`` is broken, because variable shadowing doesn't work properly.

I need to investigate further how to solve this, maybe i have to rewrite some of the Motion Solver

**Comment by jonathanplatzer** [comment]
I have found a solution to the regression, but I had to rework the `MotionEntity` class a little.
 
Do I just push the change to this branch or is there a different way this is normally done?

**Comment by jonathanplatzer** [comment]
I pushed the fix for the regression and also restored the LICENSE file of NCalc, so this should be ready to be merged.

I tested it in the 3D Motion Simulation Experiment, as well as the rewritten CRT Experiment from #509 and everything works as expected again

**Review by FlorianGlawogger — APPROVED** [review]
Thanks a lot! License file is back again, and thanks for the rewrite of MotionEntity @jonathanplatzer , the new solution is also way more readable for the default t,x,y,z,vx,vy,vz dynamic parameters 👍 also good thing that you noticed the bug!

**Pull request merged**